sys/shell/commands/sc_gnrc_icmpv6_echo.c

```diff
@@ -187,7 +187,7 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
             if (iface) {
                 data->netif = gnrc_netif_get_by_pid(atoi(iface));
             }
-            else if (gnrc_netif_highlander()) {
+            else if (gnrc_netif_numof() == 1) {
```

Comment:
`gnrc_netif_numof()` should not be used IMHO to determine if there is only one interface (due to the performance issues pointed out in #13736. Here is my solution I was about to PR, when you beat me to the race ;-):

```diff
diff --git a/sys/shell/commands/sc_gnrc_icmpv6_echo.c b/sys/shell/commands/sc_gnrc_icmpv6_echo.c
index 9dddcc8ab..d01f0711c 100644
--- a/sys/shell/commands/sc_gnrc_icmpv6_echo.c
+++ b/sys/shell/commands/sc_gnrc_icmpv6_echo.c
@@ -187,8 +187,16 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
             if (iface) {
                 data->netif = gnrc_netif_get_by_pid(atoi(iface));
             }
-            else if (gnrc_netif_highlander()) {
-                data->netif = gnrc_netif_iter(NULL);
+            else {
+                /* preliminary select the first interface */
+                gnrc_netif_t *netif = gnrc_netif_iter(NULL)
+
+                /* if there is more than one interface */
+                if (!gnrc_netif_highlander() && gnrc_netif_iter(data->netif)) {
+                    /* don't take it */
+                    netif = NULL;
+                }
+                data->netif = netif;
             }
 
             if (ipv6_addr_from_str(&data->host, data->hostname) == NULL) {
```

Reply:
Edit: fixed the netif.

Reply:
This is a shell command… I doubt the added code is providing a significant performance improvement over iterating a very short linked list.

But it provides a good opportunity to print a warning as to why no ping was sent out.

Reply:
> This is a shell command… I doubt the added code is providing a significant performance improvement over iterating a very short linked list.
> 
> But it provides a good opportunity to print a warning as to why no ping was sent out.

Still, with your change we loose the optimization, for which I don't see any reason to drop it.

Reply:
Ah no we can't print a warning here yet.

Reply:
> Still, with your change we loose the optimization, for which I don't see any reason to drop it.

One could also argue: your PR is titled "fix incorrect use of gnrc_netif_highlander()". The usage is correct, it just does not check the else case correctly ;-). My proposal keeps the usage of `gnrc_netif_highlander()` but also fixes the else case.

Reply:
I still like it because it can be factored out into a helper function:

```C
bool gnrc_netif_get(gnrc_netif_t **current)
{
    gnrc_netif_t *netif = *current;
    netif = gnrc_netif_iter(netif);

    *current = netif;
    return !gnrc_netif_highlander() && gnrc_netif_iter(netif);
}
```

Then here we would have
```patch
@@ -187,16 +187,8 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
             if (iface) {
                 data->netif = gnrc_netif_get_by_pid(atoi(iface));
             }
-            else {
-                /* preliminary select the first interface */
-                gnrc_netif_t *netif = gnrc_netif_iter(NULL)
-
-                /* if there is more than one interface */
-                if (!gnrc_netif_highlander() && gnrc_netif_iter(netif)) {
-                    /* don't take it */
-                    netif = NULL;
-                }
-                data->netif = netif;
+            else if (gnrc_netif_get(&data->netif)) {
+                data->netif = NULL;
             }
```